### PR TITLE
Porting API documentation

### PR DESCRIPTION
**Pull request by lucasponce**
- Migrate swagger annotations to Hawkular Doc annotations
- Simplify and eliminate technical debt into the docgen profile
- Clean unused files from C* implementation

This is a long PR but it was necessary as in .next we didn't have support for API documentation.
The goal of this work is to get rid of swagger annotations (it was not adding value but technical debt) and clean the documentation annotations in a way that is easier to maintain and adapt into whatever custom semantic.
Now, the -Pdocgen creates a rest-alerts.adoc under the rest-handlers/target/generated folder.

.next has not yet support for automatically update the documentation into the hawkular.org site, it is needed a previous step to refactor the current format and define a "Version 1" / "Version 2" entries to have this clear and avoid misunderstanding from future users.

### CODE REVIEW

File: api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java [hunk 1]
```diff
@@ -86,38 +85,35 @@
      * Number of conditions associated with a particular trigger.
      * i.e. 2 [ conditions ]
      */
-    @ApiModelProperty(value = "Number of conditions associated with a particular trigger. This is a read-only value " +
+    @DocModelProperty(description = "Number of conditions associated with a particular trigger. This is a read-only description " +
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java]: In the text: description -> value

File: api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java [hunk 2]
```diff
@@ -86,38 +85,35 @@
...
     protected int conditionSetSize;
 
     /**
      * Index of the current condition
      * i.e. 1 [ of 2 conditions ]
      */
-    @ApiModelProperty(value = "Index of the current condition. This is a read-only value defined by the system.",
-            position = 5,
-            required = false)
+    @DocModelProperty(description = "Index of the current condition. This is a read-only description defined by the system.",
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java]: In the text: description -> value

File: api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java [hunk 3]
```diff
@@ -86,38 +85,35 @@
...
     @JsonInclude
     protected int conditionSetIndex;
 
     /**
      * A composed key for the condition
      */
-    @ApiModelProperty(value = "A composed key for the condition. This is a read-only value defined by the system.",
-            position = 6,
-            required = false)
+    @DocModelProperty(description = "A composed key for the condition. This is a read-only description defined by the system.",
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java]: In the text: description -> value

File: api/src/main/java/org/hawkular/alerts/api/model/dampening/Dampening.java [hunk 4]
```diff
@@ -86,50 +84,50 @@
...
 
     @JsonInclude
-    @ApiModelProperty(value = "Time period in milliseconds for RELAXED_TIME, STRICT_TIME, STRICT_TIMEOUT",
+    @DocModelProperty(description = "Time period in milliseconds for RELAXED_TIME, STRICT_TIME, STRICT_TIMEOUT",
             position = 6,
             allowableValues = "> 0")
     private long evalTimeSetting;
 
-    @ApiModelProperty(value = "A composed key for the dampening. This is a read-only value defined by the system.",
+    @DocModelProperty(description = "A composed key for the dampening. This is a read-only description defined by the system.",
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/model/dampening/Dampening.java]: In the text: description -> value

File: api/src/main/java/org/hawkular/alerts/api/model/event/Event.java [hunk 5]
```diff
@@ -135,39 +134,39 @@
...
             position = 8)
     @JsonInclude(Include.NON_EMPTY)
     private Map<String, String> context;
 
-    @ApiModelProperty(value = "Tags defined by the user for this event. + \n " +
+    @DocModelProperty(description = "Tags defined by the user for this event. + \n " +
             "Events generated from <<Trigger>> will use Trigger.tags. + \n" +
             "Tags can be used as part of Event conditions expressions and criteria in finder methods. + \n" +
-            "Tag value cannot be null.",
+            "Tag description cannot be null.",
```
Comment (jshaughn) [inline api/src/main/java/org/hawkular/alerts/api/model/event/Event.java]: In the text: description -> value

File: rest-handlers/src/main/java/org/hawkular/alerts/handlers/util/ResponseUtil.java [hunk 6]
```diff
@@ -52,7 +54,10 @@
         PARAMS_PAGING = Arrays.asList(PARAM_PAGE, PARAM_PER_PAGE, PARAM_SORT, PARAM_ORDER);
     }
 
+    @DocModel(description = "Payload for a REST error response.")
     public static class ApiError {
+
+        @DocModelProperty(description = "Description of the error message.")
```
Comment (jshaughn) [inline rest-handlers/src/main/java/org/hawkular/alerts/handlers/util/ResponseUtil.java]: Maybe just, "The error message."

File: api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java (hunk 1 above)
Comment (lucasponce) [reply api/src/main/java/org/hawkular/alerts/api/model/condition/Condition.java]: Oh, good catch.
My IDE did an extra refactoring and it seems it changed some words inside text strings.
I thought I fixed them. 
Going to fix them, I guess following issues are related to this problem.

**Pull request merged**